app/src/main/java/com/example/myapplication/Analyzer.java

```diff
@@ -0,0 +1,92 @@
+package com.example.myapplication;
+
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class Analyzer {
+	//Words that have an "!" also have a form that ends with -ly
+	//Words that have an "@" also have a form that ends with -s
+	private static String[] goodWords = {"good","wonderful!","excellent","amazing","serene!",
+										"strong!","beautiful!","perfect!","fast","perfection",
+										"gorgeous!","best","charming","gracious!","better",
+										"well","accurate!","reliable!","extreme","awesome",
+										"nice!","cool","quality","effective!","efficient!",
+										"strategic!","robust!","consistent!","sufficient!"};
+	private static String[] badWords = {"bad!","terrible!","suck@","acts","crazy!",
+										"old","delicate!","intent","tragedy","awful!",
+										"vile!","tragic!","weak!","worst","traumatic!",
+										"stinging","accident!","nightmare","victim", "poor!",
+										"slow!", "unreliable!","worse","boring","trash",
+										"doesnt","cannot","inconsistent!","jittery","broken",
+										"inefficient!","horrible!","insufficient!","broke","intentional!",
+										"blind!"};
+	private static String[] multiWords = {"-0.6,not","1.8,very","0.9,most!","0.6,mild!","1.5,too",
+										"2.3,extremely","1.3,high!","1.4,completely"};
+	private static boolean populated = false;
+	public static int contains(String[] list, String str) {
+		for (int i = 0; i < list.length; i ++) {
+			if (keepLettersOnly(list[i]).equalsIgnoreCase(str)) {
+				return i;
+			}
+		}
+		return -1;
+	}
+	public static double analyze(String str) {
+		if (!populated) {
+			goodWords = populateForms(goodWords);
+			badWords = populateForms(badWords);
+			multiWords = populateForms(multiWords);
+			populated = true;
+		}
+		str = str.toLowerCase();
+		str = keepLettersOnly(str);
+		String[] words = str.split(" ");
+		double total = 0;
+		double cur = 1.0;
+		for (String s: words) {
+			int index;
+			if (contains(goodWords, s) >= 0) {
+				total += 10 * cur;
+				cur = 1.0;
+			} else if (contains(badWords, s) >= 0) {
+				total += -10 * cur;
+				cur = 1.0;
+			} else if ((index = contains(multiWords, s)) >= 0) {
+				cur *= Double.parseDouble(multiWords[index].substring(0, multiWords[index].indexOf(",")));
+			}
+		}
+		return Math.round(total * 10)/10.0;
+	}
+	public static String keepLettersOnly(String str) {
+		String[] words = str.split("");
```

Comment:
Suggestion: this method can be implemented as a single line: `return str.replaceAll("[^a-z ]", "");`

Reply:
should it include capitals letters?

Reply:
@vkirub, all letters would be turned into lowercase before it goes through the function.

Reply:
Done.